src/ExtensionConfigHandler.php

```diff
@@ -352,6 +352,156 @@ protected function getSourceStorageWrapper(array $source_dirs, $subdir = Install
     return $source_storage;
   }
 
+  public function sortAndFilterOutput($dependencies) {
+    // Sort and filter dependency lists, and finally change to simple indexed
+    // array lists. This means the output can then be directly copied for use
+    // in a .info.yml file if the output format is YAML.
+    foreach (array_keys($dependencies) as $dependency_type) {
+      if ($dependencies[$dependency_type]) {
+        sort($dependencies[$dependency_type]);
+        $dependencies[$dependency_type] = array_values($dependencies[$dependency_type]);
+      }
+      else {
+        unset($dependencies[$dependency_type]);
+      }
+    }
+    return $dependencies;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfigDependancies($extension, $limit = NULL) {
+    $dependencies = array();
+    $listed_config = $this->getExtensionInfo($extension, 'managed');
+    foreach ($listed_config as $key) {
+      $config_dependencies = $this->getIndividualConfigDependancies($key, $limit);
+      foreach ($config_dependencies as $dependency_type => $type_dependencies) {
+        if (!isset($dependencies[$dependency_type])) {
+          $dependencies[$dependency_type] = array();
+        }
+        $dependencies[$dependency_type] += $type_dependencies;
+      }
+    }
+    // Exclude 'core' and the specified extension from the full list.
+    unset($dependencies['module']['core']);
+    unset($dependencies['module'][$extension]);
+    return $this->sortAndFilterOutput($dependencies);
+  }
+
+  /**
+   * @param $name
+   *   The config key to find the dependencies of.
+   * @param null $recursion_limit
+   *   Optionally limit the levels of recursion.
+   * @return mixed
+   */
+  public function getIndividualConfigDependancies($name, $recursion_limit = NULL) {
```

Comment:
I wonder if it's worth renaming this and the getConfigDependancies/getConfigDependencies function to be clearer. This one gets dependencies for a specific item of config, the other gets dependencies for all the config in an extension. So how about changing the other to be getExtensionConfigDependencies() and then this one is actually truer to be called getConfigDependencies ?
(...and then update the interface appropriately too, so that the one that covers the whole extension is the one in the interface)

More importantly, please can we make this function protected, or if there's a good reason not to (?), add it to the interface too?